requests/models.py

```diff
@@ -400,6 +400,11 @@ def prepare_headers(self, headers):
 
         if headers:
             self.headers = CaseInsensitiveDict((to_native_string(name), value) for name, value in headers.items())
+            # Explicitly convert any non-string header value to its string
+            # representation.
+            for key, val in self.headers.items():
+                if not isinstance(val, basestring):
+                    self.headers[key] = str(val)
```

Comment:
I was thinking of doing this and then went into lazy mode. :D Let me fix it up real quick.